src/EditorFeatures/Core/Intents/DeleteParameterIntentProvider.cs

```diff
@@ -0,0 +1,75 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ChangeSignature;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Features.Intents;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Rename;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.EditorFeatures.Intents;
+
+[IntentProvider(WellKnownIntents.DeleteParameter, LanguageNames.CSharp), Shared]
+internal class DeleteParameterIntentProvider : IIntentProvider
+{
+    private readonly IGlobalOptionService _globalOptionService;
+
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public DeleteParameterIntentProvider(IGlobalOptionService globalOptionService)
+    {
+        _globalOptionService = globalOptionService;
+    }
+
+    public async Task<ImmutableArray<IntentProcessorResult>> ComputeIntentAsync(
+        Document priorDocument,
+        TextSpan priorSelection,
+        Document currentDocument,
+        IntentDataProvider intentDataProvider,
+        CancellationToken cancellationToken)
+    {
+        var changeSignatureService = priorDocument.GetRequiredLanguageService<AbstractChangeSignatureService>();
+        var contextResult = await changeSignatureService.GetChangeSignatureContextAsync(
+            priorDocument, priorSelection.Start, restrictToDeclarations: false, _globalOptionService.CreateProvider(), cancellationToken).ConfigureAwait(false);
+
+        if (contextResult is not ChangeSignatureAnalysisSucceededContext context)
+        {
+            return ImmutableArray<IntentProcessorResult>.Empty;
+        }
+
+        var parameterIndexToDelete = context.ParameterConfiguration.SelectedIndex;
+        var parameters = context.ParameterConfiguration.ToListOfParameters();
+        var isExtensionMethod = context.ParameterConfiguration.ThisParameter != null;
+
+        if (isExtensionMethod && parameterIndexToDelete == 0)
+        {
+            // We can't delete the 'this' parameter of an extension method.
+            return ImmutableArray<IntentProcessorResult>.Empty;
+        }
+
+        var newParameters = parameters.RemoveAt(parameterIndexToDelete);
+
+        var signatureChange = new SignatureChange(context.ParameterConfiguration, ParameterConfiguration.Create(newParameters, isExtensionMethod, selectedIndex: 0));
+        var changeSignatureOptionResult = new ChangeSignatureOptionsResult(signatureChange, previewChanges: false);
+
+        var changeSignatureResult = await changeSignatureService.ChangeSignatureWithContextAsync(context, changeSignatureOptionResult, cancellationToken).ConfigureAwait(false);
+        if (!changeSignatureResult.Succeeded)
+        {
+            return ImmutableArray<IntentProcessorResult>.Empty;
+        }
+
+        Contract.ThrowIfNull(changeSignatureResult.UpdatedSolution);
```

Comment:
question: is the assertion here needed? I saw the `[MemberNotNullWhen(true, nameof(UpdatedSolution))]` attribute on `Succeeded` property

Reply:
ah looks like not, good catch!